synapseclient/doze.py

```diff
@@ -0,0 +1,29 @@
+'''
+Created on Sep 21, 2017
+
+@author: bhoff
+
+sleep while checking registered listeners
+'''
+import time
+
+listeners=[]
+
+def add_listener(listener):
+    global listeners
```

Comment:
global -> nonlocal ??

Reply:
I was wrong about the use of `nonlocal` during our previous. `nonlocal` would not work because it is a Python 3 only feature and we have to also support Python 2.

Using `del listeners[:]` instead of `listeners=[]` avoids having to use `global` since you are no longer reassigning `listeners` so there would not be a scoping issue